File: src/DepositfilesSDK/DepositfilesSDK/DClient.vb
Imports DepositfilesSDK.JSON
Imports Newtonsoft.Json

Public Class DClient
    Implements IClient



    Public Sub New(Username As String, Password As String, Optional Settings As ConnectionSettings = Nothing)
        Net.ServicePointManager.Expect100Continue = True : Net.ServicePointManager.SecurityProtocol = Net.SecurityProtocolType.Tls Or Net.SecurityProtocolType.Tls11 Or Net.SecurityProtocolType.Tls12 Or Net.SecurityProtocolType.Ssl3
        If Settings Is Nothing Then
            m_proxy = Nothing
        Else
            m_proxy = Settings.Proxy
            m_CloseConnection = If(Settings.CloseConnection, True)
            m_TimeOut = If(Settings.TimeOut = Nothing, TimeSpan.FromMinutes(60), Settings.TimeOut)
        End If
        AccToken = GetToken.Token(Username, Password).Result.data.token
    End Sub
    Public Sub New(accessToken As String, Optional Settings As ConnectionSettings = Nothing)
        AccToken = accessToken
        If Settings Is Nothing Then
            m_proxy = Nothing
        Else
            m_proxy = Settings.Proxy
            m_CloseConnection = If(Settings.CloseConnection, True)
            m_TimeOut = If(Settings.TimeOut = Nothing, TimeSpan.FromMinutes(60), Settings.TimeOut)
        End If
        Net.ServicePointManager.Expect100Continue = True : Net.ServicePointManager.SecurityProtocol = Net.SecurityProtocolType.Tls Or Net.SecurityProtocolType.Tls11 Or Net.SecurityProtocolType.Tls12 Or Net.SecurityProtocolType.Ssl3
    End Sub


    Public ReadOnly Property Files(FileID As String) As IFiles Implements IClient.Files
        Get
            Return New FilesClient(FileID)
        End Get
    End Property
    Public ReadOnly Property Folders(FolderID As String) As IFolders Implements IClient.Folders
        Get
            Return New FoldersClient(FolderID)
        End Get
    End Property
    Public ReadOnly Property Root() As IRoot Implements IClient.Root
        Get
            Return New RootClient()
        End Get
    End Property

    Public ReadOnly Property RootID() As String Implements IClient.RootID
        Get
            Return "_root"
        End Get
    End Property

#Region "UserInfo"
    Public Async Function GET_UserInfo() As Task(Of JSON_GetToken) Implements IClient.UserInfo
        Using localHttpClient As New HttpClient(New HCHandler)
            Using response As Net.Http.HttpResponseMessage = Await localHttpClient.GetAsync(New pUri("user/profile", New AuthDictionary)).ConfigureAwait(False)
                Dim result As String = Await response.Content.ReadAsStringAsync()

                If CheckStatus(result) Then
                    Return JsonConvert.DeserializeObject(Of JSON_GetToken)(result, JSONhandler)
                Else
                    ShowError(result)
                End If
            End Using
        End Using
    End Function
#End Region




End Class

File: DepositfilesSDK/DepositfilesSDK/DClient.cs
using DepositfilesSDK.JSON;
using Newtonsoft.Json;
using System;
using System.Net;
using System.Threading.Tasks;
using static DepositfilesSDK.Basic;

namespace DepositfilesSDK
{
    public class DClient : IClient
    {

        public DClient(string Username, string Password, ConnectionSettings Settings = null)
        {
            ServicePointManager.Expect100Continue = true; ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls12 | SecurityProtocolType.Ssl3;
            AccToken = Authentication.Token(Username, Password).Result.data.token;
            ConnectionSetting = Settings;

            if (Settings == null)
            {
                m_proxy = null;
            }
            else
            {
                m_proxy = Settings.Proxy;
                m_CloseConnection = Settings.CloseConnection ?? true;
                m_TimeOut = Settings.TimeOut ?? TimeSpan.FromMinutes(60);
            }
        }

        public DClient(string accessToken, ConnectionSettings Settings = null)
        {
            ServicePointManager.Expect100Continue = true; ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls12 | SecurityProtocolType.Ssl3;
            AccToken = accessToken;
            if (Settings == null)
            {
                m_proxy = null;
            }
            else
            {
                m_proxy = Settings.Proxy;
                m_CloseConnection = Settings.CloseConnection ?? true;
                m_TimeOut = Settings.TimeOut ?? TimeSpan.FromMinutes(60);
            }

        }


        public IFile File(string FileID) => new FileClient(FileID);
        public IFiles Files(string[] FilesID) => new FilesClient(FilesID);
        public IFolder Folder(string FolderID)=> new FolderClient(FolderID);
        public IRoot Root() => new RootClient();


        public async Task<JSON_GetToken> UserInfo()
        {
            using (HtpClient localHttpClient = new HtpClient(new HCHandler()))
            {
                using (System.Net.Http.HttpResponseMessage response = await localHttpClient.GetAsync(new pUri("user/profile", new AuthDictionary())).ConfigureAwait(false))
                {
                    string result = await response.Content.ReadAsStringAsync();
                    return result.CheckStatus() ? JsonConvert.DeserializeObject<JSON_GetToken>(result, JSONhandler) : throw ShowError(result);
                }
            }
        }


    }
}
